src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -94,7 +94,7 @@ protected AbstractNode(TreeInfo treeInfo) {
         if (children == null) {
             children = createChildren();
         }
-        return Collections.unmodifiableList(children);
```

Comment:
Please explain why the list can not be immutable. I see no related code in this PR.

Reply:
it seems like I have changed it before I came to final version
reverted it